kie-spring-boot/kie-spring-boot-samples/kie-server-spring-boot-sample/src/test/java/org/kie/server/springboot/samples/KieServerMigrationTest.java

```diff
@@ -142,22 +142,25 @@ public void testProcessStartAndAbort() {
         assertEquals(containerId, processInstance.getContainerId());
         
         try {
-            
             MigrationReportInstance report = processAdminClient.migrateProcessInstance(containerId, processInstanceId, containerId2, processId);
             assertTrue(report.isSuccessful());
-            
             processInstance = queryClient.findProcessInstanceById(processInstanceId);
             assertNotNull(processInstance);
             assertEquals(1, processInstance.getState().intValue());
             assertEquals(containerId2, processInstance.getContainerId());
-        } finally {
-            // at the end abort process instance
-            processClient.abortProcessInstance(containerId2, processInstanceId);
-    
-            processInstance = queryClient.findProcessInstanceById(processInstanceId);
-            assertNotNull(processInstance);
-            assertEquals(3, processInstance.getState().intValue());
         }
+        catch (Exception ex ) {
+            // clean up if failure
+            try {
+                processClient.abortProcessInstance(containerId, processInstanceId);
+            } catch (Exception ex2) {}
+            throw ex;
+        }
+        //abort process instance (outside try/cath), to not hide exception that will certainly occurs if  migration failed 
+        processClient.abortProcessInstance(containerId2, processInstanceId);
```

Comment:
If the migration has failed, that code is not reached and it should not be (abort over the former container is performed). This code only makes sense in a sunny day scensario (aborting the migrated instance). If that abort failed, there is nothing else to do, since the instance is not any longer in the former container (has been properly migrated)

Reply:
got it now - thanks!